### PR TITLE
chore: prune some deps

### PR DESCRIPTION
**Pull request by Arqu**
@rklaehn Should we work to bump hyper to 1.5?

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,12 +24,11 @@ hyper = { version = "0.14.16", features = ["full"], optional = true }
 iroh-net = { version = "0.28.1", optional = true }
 pin-project = "1"
 quinn = { package = "iroh-quinn", version = "0.12", optional = true }
-serde = { version = "1.0.183", features = ["derive"] }
+serde = { version = "1.0", features = ["derive"] }
```
[the remark this answers is not in the document]
Comment (Arqu) [reply Cargo.toml:27]: Ohhh right, very good info!

[the remark this answers is not in the document]
Comment (rklaehn) [reply Cargo.toml:27]: How do you find the minimal version that works? Bisect?

[the remark this answers is not in the document]
Comment (flub) [reply Cargo.toml:27]: Yeah... basically bissect.  Because rust crates don't have changelogs.  I did this manually in beetle, maybe there's a smarter way.

**Comment by Arqu** [comment]
And here I was pulling out my hair while the minimal version was broken all along. I did move the needle forward a bit on it though.

**Pull request merged**